lib/matplotlib/path.py

```diff
@@ -493,9 +492,17 @@ def contains_point(self, point, transform=None, radius=0.0):
         """
         if transform is not None:
             transform = transform.frozen()
-        result = _path.point_in_path(point[0], point[1], radius, self,
-                                     transform)
-        return result
+        # `point_in_path` does not handle nonlinear transforms, so we
+        # transform the path ourselves.  If `transform` is affine, letting
+        # `point_in_path` handle the transform avoids allocating an extra
+        # buffer.
+        # We need to import Affine2D here to prevent a circular import between
+        # the .path and .transforms modules.
+        from .transforms import Affine2D
+        if transform and not isinstance(transform, Affine2D):
```

Comment:
Can't this be `if not transform.is_affine:` ?  That seems way better that `isinstance` anyway.